### ISSUE TITLE
Update Azure docs to v7

### ISSUE THREAD

**Issue by yvesgoeleven** [Particular/docs.particular.net#1255]
This is an epic, which covers all changes needed to the documentation in function of the next version of our azure support, v7. It references issues for Azure Storage Queues Transport, Azure Servicebus Transport, Azure Storage Persistence, Azure Blobstorage Databus and Azure Cloudservice Hosting

#### General 

- [x] Restructure content, keeping into account following remarks to be taken (also to be applied to pre v7 content if you update a document): https://github.com/Particular/docs.particular.net/issues/1151
- [x] Samples: https://github.com/Particular/docs.particular.net/issues/1014 & https://github.com/Particular/docs.particular.net/issues/804
- [ ] Snippets: https://github.com/Particular/docs.particular.net/issues/1058

#### Azure Storage Queues Transport (Covered in [beta](https://github.com/Particular/PlatformDevelopment/issues/723) release)

- [ ] ~~Upgrade guide: https://github.com/Particular/docs.particular.net/issues/1275~~
- [ ] ~~Transaction options: https://github.com/Particular/docs.particular.net/issues/1186~~
- [ ] ~~Multi namespace support: https://github.com/Particular/docs.particular.net/issues/1276~~
- [ ] ~~Native integration: https://github.com/Particular/docs.particular.net/issues/887~~
    - [ ] ~~Operational scripting: https://github.com/Particular/docs.particular.net/issues/1256~~
- [ ] ~~Keep connectionstrings safe: https://github.com/Particular/docs.particular.net/issues/1277~~
- [ ] ~~Update perf tuning https://github.com/Particular/docs.particular.net/issues/1434~~

#### Azure Servicebus Transport (Covered by [RC release](https://github.com/Particular/PlatformDevelopment/issues/753) and [Beta release](https://github.com/Particular/PlatformDevelopment/issues/718))

- ~~Upgrade guide: https://github.com/Particular/docs.particular.net/issues/1271~~
- ~~Configuration api: https://github.com/Particular/docs.particular.net/issues/1270~~
- ~~Topologies https://github.com/Particular/docs.particular.net/issues/1259~~
    - ~~Standard topology~~
    - ~~Forwarding topology~~
- [ ] ~~Transaction options: https://github.com/Particular/docs.particular.net/issues/1187~~
    - [ ] ~~Send via: https://github.com/Particular/docs.particular.net/issues/1257~~
- [ ] ~~Native integration https://github.com/Particular/docs.particular.net/issues/886~~
    - [ ] ~~Operational scripting: https://github.com/Particular/docs.particular.net/issues/894~~
    - [ ] ~~Brokered message creation: https://github.com/Particular/docs.particular.net/issues/1266~~
- ~~Multi namespace support: https://github.com/Particular/docs.particular.net/issues/368~~
- ~~MessageReceiver/MessageSender settings: https://github.com/Particular/docs.particular.net/issues/1261~~
- [ ] ~~Entity name length limitations and how to change the validation rules https://github.com/Particular/docs.particular.net/issues/1262~~
- ~~Message lock renewal: https://github.com/Particular/docs.particular.net/issues/1263~~
- ~~Batching: https://github.com/Particular/docs.particular.net/issues/1264~~
    - [ ] ~~Batch size calculation: https://github.com/Particular/docs.particular.net/issues/1273~~
- ~~Max Retries: https://github.com/Particular/docs.particular.net/issues/1265~~
- [ ] ~~Namespace hierarchy: https://github.com/Particular/docs.particular.net/issues/1267~~
- ~~keep connectionstrings safe: https://github.com/Particular/docs.particular.net/issues/1268~~
- ~~publisher names: https://github.com/Particular/docs.particular.net/issues/1269~~
- ~~MD5 vs SHA1: https://github.com/Particular/docs.particular.net/issues/1272~~
- [ ] ~~Oversized message error handling https://github.com/Particular/docs.particular.net/issues/1274~~

Crossed out items are handled in https://github.com/Particular/PlatformDevelopment/issues/718 as part of beta release or https://github.com/Particular/PlatformDevelopment/issues/753 as part of RC
 
#### Azure Storage Persistence

- ~~Connection string changes: https://github.com/Particular/docs.particular.net/pull/1243~~
- [ ] Update perf tuning https://github.com/Particular/docs.particular.net/issues/1434

#### Azure Blobstorage Databus

- [ ] Explanation tech choices: https://github.com/Particular/docs.particular.net/issues/866

#### Azure Cloudservice Hosting

- ~~CriticalError behavior: https://github.com/Particular/docs.particular.net/issues/1260~~
- [x] Configuration Changes: https://github.com/Particular/docs.particular.net/issues/1448
- [ ] Azure Host - how to run Cloud Service based endpoint without cloud service emulator #1583

**Comment by weralabaj** [comment]
@Particular/azure-maintainers Given that there's a huge number of issues in docs, is there anything we could do to help you?
Would it be possible to identify issues that don't require intimate Azure knowledge, or pair with one of you to move it forward?

**Comment by Scooletz** [comment]
In my opinion, at least the following does not require intimate Azure knowledge:
#1272 - based on https://github.com/Particular/NServiceBus.AzureStorageQueues/issues/25
#1268 - based on https://github.com/Particular/NServiceBus.AzureStorageQueues/issues/27 for ASQ

As soon as I close the saga related issue, I'd be more than happy to pair with some with docs knowledge to consume some of the issues above.

**Comment by weralabaj** [comment]
Awesome @Scooletz ping me when you're ready.

**Comment by SeanFeldman** [comment]
There are 2 types of doco we need: upgrade guides (if needed) and re-designed azure "components" (and by components I mean transports, persistence, etc). If we could have upgrade guides that would mention things and point to the "components" doco, that would be optimal. 

Example: [NHibernate upgrade guide](http://docs.particular.net/nservicebus/upgrades/nhibernate-6to7).

**Comment by weralabaj** [comment]
@Particular/azure-maintainers Given that effort for this one is huge, I think we need a proper TF formed that will get it done. 
I volunteer from the docs mainteners side, but we need at least 2-3 people that worked on Azure. To get it ready I suggest adding info on who has knowledge to tackle each issue (e.g. "Topologies #1259 ( @Scooletz) or "Topologies #1259 (anybody, refer to this and that MSDN doco)"and form the TF.

**Comment by yvesgoeleven** [comment]
@weralabaj count me in for this TF

**Comment by weralabaj** [comment]
OK one more thing I forgot - please, indicate when you can get started on this. There's still some dev work going on, right?

**Comment by yvesgoeleven** [comment]
@weralabaj I want to prioritize this over any additional dev work

**Comment by melkio** [comment]
@weralabaj I'd like to help

**Comment by SeanFeldman** [comment]
I'm in @weralabaj

**Comment by Scooletz** [comment]
@weralabaj I can't imagine not being in this TF!

**Comment by yvesgoeleven** [comment]
@Scooletz fixed :)

**Comment by yvesgoeleven** [comment]
probably best to ping @dbelcham @WilliamBZA @DavidBoike as well as they worked on the persistence part, their eyes (and eyes) will be valuable to us :)

**Comment by WilliamBZA** [comment]
Yeah, sure. I'm in. Currently involved in another TF and a few other items, so won't be able to give full commitment but I can definitely take part.

**Comment by yvesgoeleven** [comment]
@WilliamBZA added you to the list

**Comment by yvesgoeleven** [comment]
@Particular/platform-dev-squad As documentation seems to be part of the WIP limit now, I would like to nominate this issue for next in line. Documentation is the only thing missing that would allow us to release the azure related bits to our customers, and should therefore be prioritized imo. The taskforce is defined, all subtasks are identified, we are ready to go

**Comment by andreasohlund** [comment]
It's a high prio but atm getting the beta binaries prepared and fixing the rabbit message loss bug has higher prio (next in line) but this one should be a high prio after that.

**Comment by yvesgoeleven** [comment]
@andreasohlund I don't think users will do much with binaries with at least basic doco in place

**Comment by yvesgoeleven** [comment]
In short I believe we need this before we can go into beta with azure customers

**Comment by andreasohlund** [comment]
Are all doc items on this card beta blockers?

**Comment by melkio** [comment]
ASB, for instance, has a lot of new features and a lot of changes. Without documentation users can't understand well how transport works. 
Going into beta without docu in place could create issues to customers and/or a lot of questions (support cases?) to us

**Comment by andreasohlund** [comment]
I agree that this should be a prio, but non of the azure bits are in the first beta wave

https://github.com/Particular/CustomerSuccess/issues/207

so getting the binaries ready for that wave seem more important to me:

https://github.com/Particular/PlatformDevelopment/issues/706

@Particular/platform-dev-squad thoughts?

**Comment by SeanFeldman** [comment]
Upgrade guides are needed. Modification to existing documentation is needed. Especially the doco referenced from upgrade guides. And those will be required for for the "Azure wave" which will not have the time equivalent to the time invested in the general (core) doco.

**Comment by melkio** [comment]
@andreasohlund maybe unblocking upgrade guides for transports ( #1271 and #1275 ) could be a first step to start working on Azure related docu.
We can't think to start beta without them, and we can't start working on them just before beta for azure users starts

**Comment by weralabaj** [comment]
@andreasohlund how about putting perf tests on hold for a few days until we get free slot? I think this is crucial to get before beta (at least some parts of it) and it'll take a while to get done.

**Comment by weralabaj** [comment]
Scratch it - https://github.com/Particular/PlatformDevelopment/issues/673#issuecomment-206185305 - is it true that neither Azure nor Rabbit are part of beta?

**Comment by andreasohlund** [comment]
Not part of private beta phase one, but part of the public beta for the entire platform so they are part of that's the next focus once the phase 1 binaries are up

**Comment by WilliamBZA** [comment]
Stop-starting other issues to bump this into WIP is a less-than-ideal solution and I'd much rather prefer to wait for a WIP slot to open because of the completion of a previous task. If there are tasks in progress that aren't critical to V6, that highlights a problem with the WIP selection process.

If we stop another task to start this one, there is no guarantee that the previous task will still be deemed as important enough to start again once another slot opens up.

**Comment by weralabaj** [comment]
Good point @WilliamBZA, sorry ;) I think we have two candidates for resolving soon and we should focus on them today.

**Comment by andreasohlund** [comment]
>If there are tasks in progress that aren't critical to V6, that highlights a problem with the WIP selection process.

Is there anything in-progress that you would say fits this description?

**Comment by weralabaj** [comment]
3614 and 68?

**Comment by andreasohlund** [comment]
Issue raised for ASB raised https://github.com/Particular/PlatformDevelopment/issues/718, I've proposed that it will be next in line (issues for ASQ and storage etc will follow)

**Comment by SeanFeldman** [comment]
Updated ASB section to reflect what items will be handled as part of the ASB Beta [issue #718](https://github.com/Particular/PlatformDevelopment/issues/718) in PlatDev.

**Comment by andreasohlund** [comment]
Removing the task force from this one, all work will be done/tracked as part of 

https://github.com/Particular/PlatformDevelopment/issues/718
https://github.com/Particular/PlatformDevelopment/issues/723
https://github.com/Particular/PlatformDevelopment/issues/729


and upcoming issue to get the above RC ready. We'll keep this one open until all items here are assigned to a relevant plat dev issue.

**Comment by indualagarsamy** [comment]
ping @Particular/azure-host-maintainers @Particular/azure-service-bus-maintainers @Particular/azure-storage-persistence-maintainers  @Particular/azure-storage-queues-maintainers 

Any updates or progress?

**Comment by SeanFeldman** [comment]
A few minor items are not finished yet. I suspect they are either WIP or done, but not tracked.

**Comment by SeanFeldman** [comment]
Looks like we can close this "umbrella" issue.
In case individual issues are not addressed, they can be tackle independently by maintainers.
